v2/src/log_hresult.c

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <inttypes.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "windows.h"
+#include "psapi.h"
+
+#include "c_logging/format_message_no_newline.h"
+#include "c_logging/log_hresult.h"
+
+#define MESSAGE_BUFFER_SIZE 512
+#define N_MAX_MODULES 10
+
+static const char snprintf_failure_message[] = "snprintf failed";
+
+int log_hresult_fill_property(void* buffer, HRESULT hresult)
+{
+    if (buffer == NULL)
+    {
+        /* Codes_SRS_LOG_HRESULT_01_001: [ If buffer is NULL, log_hresult_fill_property shall return 512 to indicate how many bytes shall be reserved for the HRESULT string formatted version. ] */
+    }
+    else
+    {
+        /*see if the "system" can provide the code*/
+        /* Codes_SRS_LOG_HRESULT_01_002: [ log_hresult_fill_property shall call FormatMessageA_no_newline with FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, hresult, 0 as language Id, buffer as buffer to place the output and 512 as buffer size. ] */
```

Comment:
>512 [](http://example.com/codeflow?start=239&length=3)

this 512 buffer size seems to be "by convention" passed around and assumed. 


Any way it can be made explicit as an argument?

Reply:
I'll add a task for it. It could be, but then it would require more changes in log_context, which I'd rather do separately.

ID Work Item Type Title State Assigned To Effort Remaining Work Original Estimate Value Area Iteration Path Tags
24248404 Task Functions used by LOG_CONTEXT_PROPERTY_CUSTOM_FUNCTION should use explicit buffer sizes rather than by convention To Do           One